inception-external-search-solr/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/solr/SolrSearchProvider.java

```diff
@@ -0,0 +1,330 @@
+     * @param document contain all the information about the document
+     * @param aTraits request parameters
+     */
+    private void fillResultWithMetadata(ExternalSearchResult result, SolrDocument document,
+                                        SolrSearchProviderTraits aTraits)
+            throws IOException, SolrException
+    {
+        if (isNotBlank((String) document.getFirstValue(DOC_NAME_KEY)))
+        {
+            result.setDocumentTitle((String) document.getFirstValue(DOC_NAME_KEY));
+        }
+        else if (isNotBlank((String) document.getFirstValue(DOC_TITLE_KEY)))
+        {
+            result.setDocumentTitle((String) document.getFirstValue(DOC_TITLE_KEY));
+        }
+        else
+            result.setDocumentTitle((String) document.getFirstValue(DOC_ID_KEY));
+
+        if (isNotBlank((String) document.getFirstValue(DOC_LANGUAGE_KEY)))
+            result.setLanguage((String) document.getFirstValue(DOC_LANGUAGE_KEY));
+
+
+        if (isNotBlank((String) document.getFirstValue(DOC_URI_KEY)))
+            result.setOriginalUri((String) document.getFirstValue(DOC_URI_KEY));
+        else
+            result.setOriginalUri(aTraits.getIndexName());
+
+        // If there is no timestamp then we use the computer timestamp
+        if (isNotBlank((String) document.getFirstValue(DOC_TIMESTAMP_KEY)))
+            result.setTimestamp((String) document.getFirstValue(DOC_TIMESTAMP_KEY));
+        else
+        {
+            DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss");
+            LocalDateTime now = LocalDateTime.now();
+            result.setTimestamp(dtf.format(now));
+        }
+
+        // Field with the entire text
+        result.setOriginalSource((String) document.getFirstValue(aTraits.getTextField()));
+
+    }
+
+
+    /**
+     * Search a Solrdocument by id and return an ExternalSearchResult.
+     * @param aRepository Solr repository
+     * @param aTraits request parameters
+     * @param aCollectionId the name of the collection
+     * @param aDocumentId the id of the document
+     * @return is used by the module external search core in order to get a preview of the document
+     * @throws SolrException Connection exception and request exception
+     */
+    @Override
+    public ExternalSearchResult getDocumentResult(DocumentRepository aRepository,
+                                                  SolrSearchProviderTraits aTraits,
+                                                  String aCollectionId,
+                                                  String aDocumentId)
+            throws IOException, SolrException {
+        if (!aCollectionId.equals(aTraits.getIndexName())) {
+            throw new IllegalArgumentException(
+                    "Requested collection name does not match connection collection name");
+        }
+
+        SolrQuery getQuery = new SolrQuery(DOC_ID_KEY + ":" + aDocumentId);
```

Comment:
Is it necessary / possible to escape potential special characters in the document ID to not break the query?